src/pages/index.js

```diff
@@ -20,88 +64,191 @@ const cardSection = new Section(
+      console.log('Changed profile submit button to saving...')
+      userInfo.setUserInfo(formData);
+      profilePopup.close(); 
+    })
+    .catch((err) => {
+      console.error("Error updating profile:", err);
+    })
+    .finally(()=>{
+      profileSubmitButton.textContent = originalText;
+      console.log('changing prof submit button text back to original')
+    })
 });
+
 profilePopup.setEventListeners();
 
+
 const addCardPopup = new PopupWithForm("#modal__add", (formData) => {
+const originalText = addCardPopupSubmit.textContent;
+
   const newCardData = {
     name: formData.name.trim(),
     link: formData.url.trim(),
   };
 
   if (newCardData.name && newCardData.link) {
-    cardSection.addItem(createCard(newCardData));
-
-    // Only reset form on submit
-    document.querySelector("#add__modal_form").reset();
-
-    // Close popup after adding a card
-    addCardPopup.close();
+    api.updateNewCards(newCardData) 
+      .then((savedCard) => {
+        submitButtonTextChanger(addCardPopupSubmit);
+        // Ensure the savedCard has all necessary properties
+        const fullCardData = {
+          ...savedCard,
+          likes: [], // Initialize empty likes array
+          _id: savedCard._id, // Ensure _id is present
+        };
+
+        const cardElement = createCard(fullCardData)
+        cardSection.addItem(cardElement); 
+        addModalForm.reset();
+        addCardPopup.close(); 
+      })
+      .catch((err) => {
+        console.error("Error adding new card:", err);
+      })
+      .finally(()=>{
+        addCardPopupSubmit.textContent = originalText;
+        //to test the submitButtonChanger please 
+        //remove the code on line 160 and you'll see the text is changed 
+        console.log('changing add submit button text back to original')
+      })
   }
 });
+addCardPopup.setEventListeners();
 
+const avatarPopup = new PopupWithForm("#avatar__modal", (formData) => {
+  console.log("Avatar Form Data:", formData);
+  const originalText = avatarPopupSubmit.textContent;
+  
+
+  api.updateAvatar(formData.avatar_url) 
+    .then((updatedData) => {
+      userInfo.setAvatar(updatedData); // Updates avatar 
+      submitButtonTextChanger(avatarPopupSubmit);
+      document.querySelector("#avatar__modal_form").reset(); 
+      avatarPopup.close(); 
+    })
+    .catch((err) => {
+      console.error("Error updating avatar:", err);
+    })
+    .finally(()=>{
+      avatarPopupSubmit.textContent = originalText;
+      console.log('avatar submit button returned to normal')
+    })
+});
+avatarPopup.setEventListeners();
+
+const confirmationPopup = new PopupWithConfirm("#confirmation__modal");
+confirmationPopup.setEventListeners(); 
+
+
+function handleConfirmPopup(card, cardElement) {
+  const confirmPopupOriginal = confirmPopupSubmit.textContent;
+  confirmationPopup.setSubmitFunction(() => {
+    api.deletingCards(card)
+      .then(() => { 
+        submitButtonTextChanger(confirmPopupSubmit);
```

Comment:
`submitButtonTextChanger(confirmPopupSubmit);`  starts the loading effect, so it should be **before** the request


But it should not be `Saving..` when you **delete** a card. It should be `Deleting...`